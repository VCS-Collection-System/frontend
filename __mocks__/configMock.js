export const getConfig = (URL, configUrl, config) => {
  if(URL.indexOf(configUrl) > -1 ) {
    const country = URL.substring(configUrl.length);
    const countryConfig = config[country] || null;
    return Promise.resolve({data: countryConfig});
  } else {
    return Promise.reject({data: null});
  }
}