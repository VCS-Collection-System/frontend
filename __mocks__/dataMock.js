export const employeeInfo = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth":"2021-10-07",
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const attachmentObject = {
  "originalFileName":"blob",
  "size":100,
  "contentType":"image/png",
  "s3BucketName":"mockBucket",
  "s3UUID":"mockUUID"
}

export const vaxHistory = {
  "id":0,
  "employee": employeeInfo,
  "submissionDate":"2021-10-04T15:12:14.875",
  "review":{
     "reviewerEmployeeId":"z2",
     "reviewerNotes":"",
     "reviewDate":"2021-10-05T14:04:29.296",
     "reviewOutcome":"ACCEPTED",
     "rejectReason":""
  },
  "submittedBy":null,
  "autoApproved":false,
  "lastUpdatedTime":null,
  "lastUpdatedBy":null,
  "vaccineBrand":"MODERNA",
  "vaccineAdministrationDate":"2021-10-04",
  "fullVaccinatedFlag":false,
  "vaccineShotNumber":2,
  "documentType":"VaccineDocument",
  "attachment": {}
}

const taskEmployeeObject = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth": {
    "year": "2021",
    "monthValue": "12",
    "dayOfMonth": "16"
  },
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const mockTask = {
  "task-input-data": {
    "documentList": {
      "com.redhat.vcs.model.VaccineDocumentList": {
        "documents": [ {
          "com.redhat.vcs.model.VaccineCardDocument": {
            "employee": {
              "com.redhat.vcs.model.Employee": taskEmployeeObject
            },
            "vaccineBrand": {
              "com.redhat.vcs.model.VaccineBrand": "MODERNA"
            },
            "attachment": {
              "com.redhat.vcs.model.Attachment": attachmentObject
            },
            "vaccineAdministrationDate": {
              "year": "2021",
              "monthValue": "12",
              "dayOfMonth": "16"
            },
            "lotNumber": "mockLot"
          } }
        }
      }
    }
  }
}
