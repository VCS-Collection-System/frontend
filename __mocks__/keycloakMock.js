module.exports = {
    token: "bogustoken", 
    authenticated: false,
    loadUserProfile: jest.fn( ()=> {return Promise.resolve({
        attributes: {
            workforceid: ["mockId"]
        }
    })} ),
    login: jest.fn(),
    logout: jest.fn(),
    hasRealmRole: jest.fn(),
    profile: {
        firstName: "mockFirst",
        lastName: "mockLast",
        email: "mock@email.com",
        attributes: {
            workforceid: ["mockId"],
            agency: ["USA"]
        }
    }
};
