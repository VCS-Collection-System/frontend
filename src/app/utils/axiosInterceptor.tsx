import axios from "axios";

const axiosInterceptor = axios.create({})

axiosInterceptor.interceptors.request.use(
    config => {
        config.headers['Authorization'] = 'Bearer ' + sessionStorage.getItem("token");
        return config;
    },
    error => {
        Promise.reject(error)
    });
export default axiosInterceptor;