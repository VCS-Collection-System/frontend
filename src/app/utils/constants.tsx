declare global {  
  interface Window {
    REACT_APP_PAM_URL_BASE: string;
    REACT_APP_PAM_CONTAINER_ID: string;
    REACT_APP_KEYCLOAK_COUNTRY: string;
    REACT_APP_KEYCLOAK_EMAIL: string;
    REACT_APP_KEYCLOAK_FIRST_NAME: string;
    REACT_APP_KEYCLOAK_ID: string;
    REACT_APP_KEYCLOAK_LAST_NAME: string;
    REACT_APP_MAINTENANCE_BANNER: boolean;
    REACT_APP_MAINTENANCE_BANNER_TEXT: string;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    REACT_APP_FEATURE_FLAGS: Array<Record<string, any>>;
    REACT_APP_MAX_DOSES: Record<string, number>;
    REACT_APP_CONSENT_LABEL: string;
    REACT_APP_ALTERNATE_CONSENT_LABEL: string;
    REACT_APP_BLOCKED_COUNTRIES: Array<string>;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    REACT_APP_GLOBAL_CONFIG: Record<string, any>;
  }
}

// PAM urls
export const PAM_BASE_URL = window.REACT_APP_PAM_URL_BASE; 
export const SUBMIT_VAX_URL = PAM_BASE_URL + '/attestation/vax';
export const SUBMIT_TEST_URL = PAM_BASE_URL + '/attestation/covid-test-result';
export const PAM_TASK_URL = PAM_BASE_URL + '/tasks/';
export const RESERVED_TASK_URL = PAM_TASK_URL + 'owners?status=Reserved&user=';
export const INPROGRESS_TASK_URL = PAM_TASK_URL + 'owners?status=InProgress&user=';
export const ALL_TASK_URL = PAM_TASK_URL + 'owners?status=Ready&pageSize=200';
export const PRIORITY_URL = PAM_BASE_URL + '/query/priority-results';
export const PRIORITY_CONFIRM_URL = PAM_BASE_URL + '/query/priority-result/';
export const ATTACHMENT_URL = PAM_BASE_URL + '/attestation/attachment';
export const EMPLOYEE_URL = PAM_BASE_URL + '/query/employee/';
export const EMPLOYEE_NAME_URL = PAM_BASE_URL + '/query/employee-name/';
export const EMPLOYEE_VAX_HISTORY_URL = PAM_BASE_URL + '/query/vax-document/';
export const EMPLOYEE_TEST_HISTORY_URL = PAM_BASE_URL + '/query/test-result-document/';
export const VAX_TASK_DOCUMENT_URL = PAM_BASE_URL + '/query/task/document/';
export const DELETE_VAX_URL = PAM_BASE_URL + '/attestation/delete-vax-record/';
export const CONFIG_URL = PAM_BASE_URL + '/query/country-configs/';
export const SAVE_CONFIG_URL = PAM_BASE_URL + '/attestation/country-configs';

// List of countries allow to use this app
export const BLOCKED_COUNTRIES = window.REACT_APP_BLOCKED_COUNTRIES || []

export const KONAMI_CODE = [ "ArrowUp", "ArrowUp", "ArrowDown", "ArrowDown", "ArrowLeft", "ArrowRight", "ArrowLeft", "ArrowRight", "KeyB", "KeyA"];

// Maintenance Banner
export const MAINTENANCE_BANNER = window.REACT_APP_MAINTENANCE_BANNER || false
export const MAINTENANCE_BANNER_TEXT = window.REACT_APP_MAINTENANCE_BANNER_TEXT || ""

// Keycloak claims
export const KEYCLOAK_COUNTRY = window.REACT_APP_KEYCLOAK_COUNTRY
export const KEYCLOAK_EMAIL = window.REACT_APP_KEYCLOAK_EMAIL
export const KEYCLOAK_FIRST_NAME = window.REACT_APP_KEYCLOAK_FIRST_NAME
export const KEYCLOAK_ID = window.REACT_APP_KEYCLOAK_ID
export const KEYCLOAK_LAST_NAME = window.REACT_APP_KEYCLOAK_LAST_NAME
export const FEATURE_FLAGS = window.REACT_APP_FEATURE_FLAGS || []

// Maximum doses allowed by vaccine manufacturer
export const MAX_DOSES = window.REACT_APP_MAX_DOSES || {}

export const CONSENT_LABEL = window.REACT_APP_CONSENT_LABEL
export const ALTERNATE_CONSENT_LABEL = window.REACT_APP_ALTERNATE_CONSENT_LABEL

// Global country config
export const GLOBAL_CONFIG = window.REACT_APP_GLOBAL_CONFIG || {};

