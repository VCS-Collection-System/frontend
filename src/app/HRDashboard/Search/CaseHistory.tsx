import React, { useEffect, useState } from 'react';
import { 
  Accordion,
  AccordionContent,
  AccordionItem,
  AccordionToggle,
  Button,
  Card, 
  CardBody, 
  CardFooter,
  Grid,
  Flex,
  FlexItem,
  GridItem,
  Modal, 
  ModalVariant,
  Menu, 
  MenuContent, 
  MenuList, 
  MenuItem,
  MenuGroup,
  Split
} from '@patternfly/react-core';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useHistory } from 'react-router-dom';
import authaxios from "@app/utils/axiosInterceptor";
import { HRVaxSubmit } from './AddRecord/HRVaxSubmit';
import { HRTestSubmit } from './AddRecord/HRTestSubmit';
import { dateConvert } from "@app/utils/utils";
import EmployeeInfoHorizontalCard from '@app/Components/EmployeeInfoHorizontalCard';
import { CaseReviewPanel } from './CaseReviewPanel';
import { EMPLOYEE_VAX_HISTORY_URL, EMPLOYEE_TEST_HISTORY_URL } from '@app/utils/constants'

const CaseHistory: React.FunctionComponent = () => {
  const history = useHistory();
  const [isModalOpen, setIsModalOpen] = useState(false); 
  const [employeeId, setEmployeeId] = useState("");
  const [employeeName, setEmployeeName] = useState("");
  const [employeeDOB, setEmployeeDOB] = useState("");
  const [employeeEmail, setEmployeeEmail] = useState("");
  const [refresh, setRefresh] = useState(false);
  const [activeItem, setActiveItem] = useState('0'); 
  const [vaxHistory, setVaxHistory] = useState([]);
  const [testHistory, setTestHistory] = useState([]);  
  const [hasSelectedItem, setHasSelectedItem] = useState(false); 
  const [updatePanel, setUpdatePanel] = useState(false);
  const [expanded, setExpanded] = useState('');

  useEffect(() => {
    const sessionEmployee = sessionStorage.getItem("searchResultInfo") || "{}"; 
    const employeeInfo = JSON.parse(sessionEmployee); 
    setEmployeeId(employeeInfo['employeeId']);
    setEmployeeName(employeeInfo['firstName'] + " " + employeeInfo['lastName']);
    setEmployeeDOB(employeeInfo['dateOfBirth']);
    setEmployeeEmail(employeeInfo['email']);

    authaxios.get(EMPLOYEE_VAX_HISTORY_URL + employeeInfo['employeeId'])
    .then(result => {
      result['data'].sort((a, b) => {
        return a['vaccineAdministrationDate'].localeCompare(b['vaccineAdministrationDate']);
      });
      setVaxHistory(result['data'])
    })

    authaxios.get(EMPLOYEE_TEST_HISTORY_URL + employeeInfo['employeeId'])
    .then(result => {
      setTestHistory(result['data'])
    })

  }, [refresh]);

  const closeModal = () => {
    setIsModalOpen(false);
    setRefresh(!refresh);
    setUpdatePanel(!updatePanel);
  }

  const onSelect = (event, itemId) => {
    setHasSelectedItem(true);
    setActiveItem(itemId);
    setUpdatePanel(!updatePanel)

    if(itemId.includes('TEST')){
      const dataIndex = parseInt(itemId.split(':')[1]);
      const testData = testHistory[dataIndex];
      sessionStorage.setItem("caseHistoryType", "TEST");
      sessionStorage.setItem("caseHistoryInfo", JSON.stringify(testData));
    }
    else if(itemId.includes('VAX')) {
      const dataIndex = parseInt(itemId.split(':')[1]);
      const vaxData = vaxHistory[dataIndex];
      sessionStorage.setItem("caseHistoryType", "VAX");
      sessionStorage.setItem("caseHistoryInfo", JSON.stringify(vaxData));
    }
  }

  const getTaskStatus = (value) => {
    if(value['review']) {
      return value['review']['reviewOutcome'];
    }
    return value['autoApproved'] ? "AUTO-ACCEPTED" : "NOT REVIEWED";
  }

  const AddRecordModal = () => {
    function onToggle(id) {
      if(id === expanded) {
        setExpanded('');
      }
      else {
        setExpanded(id); 
      }
    }


    return (
       <Modal
        aria-label="AddRecordModal"
        variant={ModalVariant.medium}
        title="Add Record"
        isOpen={isModalOpen}
        onClose={closeModal}
      >
        <Accordion>
          <AccordionItem>
            <AccordionToggle
              onClick={() => {
                onToggle('vax-toggle');
              }}
              isExpanded={expanded === 'vax-toggle'}
              id="vax-toggle"
            >
              Vaccine Submission
            </AccordionToggle>
            <AccordionContent id="vax-expand" isHidden={expanded !== 'vax-toggle'}>
              <HRVaxSubmit closeParentModal={closeModal}/>
            </AccordionContent>
          </AccordionItem>
          <AccordionItem>
            <AccordionToggle
              onClick={() => {
                onToggle('test-toggle');
              }}
              isExpanded={expanded === 'test-toggle'}
              id="test-toggle"
            >
              Test Submission
            </AccordionToggle>
            <AccordionContent id="test-expand" isHidden={expanded !== 'test-toggle'}>
              <HRTestSubmit closeParentModal={closeModal}/>
            </AccordionContent>
          </AccordionItem>
        </Accordion>
        <br />
        <Flex>
          <FlexItem align={{ default: "alignLeft" }}>
            <Button key="back" variant="plain" onClick={closeModal}>
              Back
            </Button>
          </FlexItem>
        </Flex>
      </Modal>
    )
  }

  const CovidTestHistory = () => {
    if(testHistory.length > 0) {
      return (
        <>
          <MenuItem 
            key={"TEST"} 
            itemId={"TEST"}
            isDisabled
          >
            <Flex justifyContent={{default: "justifyContentSpaceBetween"}}>
              <FlexItem>
              {"Test Date"}
              </FlexItem>
              <FlexItem>
              {"Test Result"}
              </FlexItem>
            </Flex>
          </MenuItem>
          {
            testHistory.map((x, i) => (
              <MenuItem 
                key={"TEST:" + i} 
                itemId={"TEST:"+i}
              >
                <Flex justifyContent={{default: "justifyContentSpaceBetween"}}>
                  <FlexItem>
                    {dateConvert(x['covidTestDate'])}
                  </FlexItem>
                  <FlexItem>
                  { x['covidTestResult'] }
                    {/* { x['covidTestResult'] === "POSITIVE" ? <CircleIcon color='firebrick'/>  : x['covidTestResult'] === "NEGATIVE" ?  <CircleIcon color="seagreen" />: <CircleIcon color="goldenrod"/> } */}
                  </FlexItem>
                </Flex>
              </MenuItem>
            ))
          }
        </>
      )
    }
    else{
      return (
        <MenuItem key="empty" itemId="empty" isDisabled>
          [No Test Results]
        </MenuItem>
      )
    }
  }

  const CovidVaccineHistory = () => {
    if(vaxHistory.length > 0) {
      return (
        <>
          <MenuItem 
            key={"VAX"} 
            itemId={"VAX"}
            isDisabled
          >
            <Flex justifyContent={{default: "justifyContentSpaceBetween"}}>
              <FlexItem>
                {"Vaccination Date"}
              </FlexItem>
              <FlexItem>
                {"Status"} 
              </FlexItem>
            </Flex>
          </MenuItem>
          {
            vaxHistory.map((value, index) => (
              <MenuItem 
                key={"VAX:" + index} 
                itemId={"VAX:" + index}
              >
                <Flex justifyContent={{default: "justifyContentSpaceBetween"}}>
                  <FlexItem>
                    { dateConvert(value['vaccineAdministrationDate']) }
                  </FlexItem>
                  <FlexItem>
                    { getTaskStatus(value) }
                  </FlexItem>
                </Flex>
              </MenuItem>
            ))
          }
        </>
      )
    }
    else{
      return (
        <MenuItem key="empty" itemId="empty" isDisabled>
          [No Vaccination Records]
        </MenuItem>
      )
    }
  }

  return (
    <Card isRounded>
      <AddRecordModal />
      <CardWithTitle title="Case History" info="Review Case History"/>
      <CardBody>
        <Grid hasGutter>
          <GridItem span={12}> 
            <EmployeeInfoHorizontalCard 
              employeeId={employeeId}
              employeeName={employeeName}
              employeeDOB={employeeDOB}
              employeeEmail={employeeEmail}
            />
          </GridItem>
          <GridItem span={4} rowSpan={6}> 
            <Card>
              <CardWithTitle title="Attestation Records"/>
                <Menu onSelect={onSelect} activeItemId={activeItem}>
                  <MenuContent menuHeight="875px">
                    <MenuGroup label="Test Results">
                      <MenuList className="indentList">
                        <CovidTestHistory />
                      </MenuList>
                    </MenuGroup>
                    <MenuGroup label="Vaccination Records">
                      <MenuList className="indentList">
                        <CovidVaccineHistory />
                      </MenuList>
                    </MenuGroup>
                  </MenuContent>
                </Menu>
            </Card>
          </GridItem>
          <GridItem span={8} rowSpan={6}>
            <CaseReviewPanel 
              hasSelectedItem={hasSelectedItem}
              setHasSelectedItem={setHasSelectedItem}
              updatePanel={updatePanel} 
              refreshCaseHistory={refresh}
              setResfreshCaseHistory={setRefresh}
            />
          </GridItem>
        </Grid>
      </CardBody>
      <CardFooter>
        <Split hasGutter>
          <Button variant="primary" onClick = { () => setIsModalOpen(!isModalOpen) }>
            Add Record
          </Button>
          <Button 
            variant="plain" 
            onClick={() => history.push("/hrdashboard")
          }>
            Back
          </Button>
        </Split>
      </CardFooter>
    </Card>
  );
}

export { CaseHistory };