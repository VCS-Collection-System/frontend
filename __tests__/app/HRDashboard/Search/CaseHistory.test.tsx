import React from 'react';
import { CaseHistory } from '../../../../src/app/HRDashboard/Search/CaseHistory';
import mockKeycloak from '../../../../__mocks__/keycloakMock'
import { act } from 'react-dom/test-utils';
import { EMPLOYEE_VAX_HISTORY_URL, ATTACHMENT_URL } from '../../../../src/app/utils/constants'
import { fireEvent, render, screen } from '@testing-library/react'
import reactRouterDom from 'react-router-dom'
import { employeeInfo, vaxHistory,} from '../../../../__mocks__/dataMock'

let mockVaxHistory = [{}];
const errorState = 'error';

jest.mock('react-router-dom')
const mockPush = jest.fn()
reactRouterDom.useHistory = jest.fn().mockReturnValue({push: mockPush})

jest.mock('@react-keycloak/web', () => ({
    useKeycloak: jest.fn( ()=>({keycloak: mockKeycloak}))
})) 

jest.mock("@app/utils/constants", () => ({
  KEYCLOAK_FIRST_NAME: "firstName",
  KEYCLOAK_LAST_NAME: "lastName",
  KEYCLOAK_EMAIL: "email",
  KEYCLOAK_COUNTRY: "attributes.agency[0]",
  KEYCLOAK_ID: "attributes.workforceid[0]",
  APPROVER_ROLES: ["mock-approver"],
  FEATURE_FLAGS: ["fakeFeature"],
  CONSENT_COUNTRIES: ["AUS"],
  GLOBAL_CONFIG: {"default": {
    "allowPdf": false,
    "dateOfBirth": true,
    "consentCheckbox": false,
    "covidTest": false,
    "customInput": true,
    "hrdashboard": true,
    "hrsearch": true,
    "proofType": {"cdc": false, "divoc": false, "greenPass": false},
    "recovery": false,
    "fakeFeature": true
  }}
}))

jest.mock('@app/utils/axiosInterceptor', () => ({
  get: (URL) => {
    let returnData; 

    if(URL === EMPLOYEE_VAX_HISTORY_URL + employeeInfo['employeeId']){
      returnData = {data: mockVaxHistory}
    }
    else{
      returnData = {data: []};
    }

    return Promise.resolve(returnData)
  },
  post: (URL) => {
    let returnData; 

    if(URL === ATTACHMENT_URL){
      returnData = {data: {}, headers: {}};
    }

    return Promise.resolve(returnData)
  }
}))

beforeEach(() => {
  mockVaxHistory = [];
  sessionStorage.setItem("searchResultInfo", JSON.stringify(employeeInfo))
})

afterEach(() => {
  sessionStorage.clear(); 
})

describe('<CaseHistory />', ()=> {
  it('should be defined', () => {
    expect(CaseHistory).toBeDefined()
  })

  it("should match existing snapshot", async () => {
    await act(async () => {
      const { asFragment } = render(<CaseHistory />)
      expect(asFragment()).toMatchSnapshot()
    })
  })

  it("should render without Vax Data", async () => {
    await act(async () => {
      render(<CaseHistory />)
    })
    expect(screen.getByText('[No Vaccination Records]')).toBeTruthy(); 
    expect(screen.queryAllByText('Mock Tester')).toEqual([]);
    expect(screen.queryAllByText('test@mock.com')).toEqual([]);
    expect(screen.queryAllByText('mock1')).toEqual([]);
    expect(screen.queryAllByText('"2021-10-07"')).toEqual([]);
  })

  it("should render Vax Data", async () => {
    mockVaxHistory = [vaxHistory]
    await act(async () => {
      render(<CaseHistory />)
    })
    expect(screen.getByRole('button', {name: /10-04-2021 ACCEPTED/i})).toBeTruthy();
  })

  it("should be able to click Vaccination History menu items", async () => {
    mockVaxHistory = [vaxHistory]
    window.URL.createObjectURL = jest.fn();
    await act(async () => {
      render(<CaseHistory />)
    })
    const vaxItem = await screen.getByRole('button', {name: /10-04-2021 ACCEPTED/i})
    await act(async () => {
      fireEvent.click(vaxItem)
    })
    const caseObject = JSON.parse(sessionStorage.getItem("caseHistoryInfo") || "{}")
    expect(sessionStorage.getItem("caseHistoryType")).toEqual("VAX")
    expect(caseObject).toMatchObject(vaxHistory)
  })

  it("should be able render Vaccination History without review and auto approved", async () => {
    mockVaxHistory = [{
      "review": null,
      "autoApproved":true,
      "vaccineBrand":"MODERNA",
      "vaccineAdministrationDate":"2021-10-04",
      "vaccineShotNumber":2,
      "documentType":"VaccineDocument"
    }]
    await act(async () => {
      render(<CaseHistory />)
    })
    expect(screen.getByText('AUTO-ACCEPTED')).toBeTruthy()
  })

  it("should be able render Vaccination History without review and not reviewed", async () => {
    mockVaxHistory = [{
      "review": null,
      "autoApproved":false,
      "vaccineBrand":"MODERNA",
      "vaccineAdministrationDate":"2021-10-04",
      "vaccineShotNumber":2,
      "documentType":"VaccineDocument"
    }]
    await act(async () => {
      render(<CaseHistory />)
    })
    expect(screen.getByText('NOT REVIEWED')).toBeTruthy()
  })

  it("should sort Vaccination History by date", async () => {
    mockVaxHistory = [{
      "review": null,
      "autoApproved":false,
      "vaccineBrand":"MODERNA",
      "vaccineAdministrationDate":"2021-10-05",
      "vaccineShotNumber":2,
      "documentType":"VaccineDocument"
    },
    {
      "review": null,
      "autoApproved":false,
      "vaccineBrand":"MODERNA",
      "vaccineAdministrationDate":"2021-10-04",
      "vaccineShotNumber":1,
      "documentType":"VaccineDocument"
    }]
    await act(async () => {
      render(<CaseHistory />)
    })
    expect(screen.getByText((content, element) => {
      if(!element) {
        return false;
      }
      const listItem = element.closest(".pf-c-menu__list-item");
      if(!listItem) {
        return false;
      }
      return (content === "10-04-2021" &&
        listItem.nextSibling != null)
    })).toBeTruthy();
  })

  it("should be able to click Back to return to HR Dashboard", async () => {
    await act(async () => {
      render(<CaseHistory />)
    })
    const backButton = await screen.getByRole('button', {name: /Back/i})
    fireEvent.click(backButton)
    expect(mockPush).toHaveBeenCalledWith('/hrdashboard')
  })

  // update with new modal
  /*it("should be able to click Add Record to see Modal", async () => {
    await act(async () => {
      render(<CaseHistory />)
    })
    const addRecordButton = await screen.getByRole('button', {name: /Add Record/i})
    fireEvent.click(addRecordButton)
    expect(screen.getByRole('button', {name: /Close/i})).toBeTruthy();
    expect(screen.getByRole('heading', {name: /Add Record/i})).toBeTruthy();
    expect(screen.getByRole('button', {name: /Submit/i})).toBeTruthy();
  })

  it("should be able to close Add Record", async () => {
    const spy = jest.spyOn(console, errorState).mockReturnValue()
    await act(async () => {
      render(<CaseHistory />)
    })
    const addRecordButton = await screen.getByRole('button', {name: /Add Record/i})
    fireEvent.click(addRecordButton)
    const closeButton = await screen.getByRole('button', {name: /Close/i})
    fireEvent.click(closeButton)
    expect(screen.queryByText('Vaccine Submission')).toBeNull();
    expect(screen.queryByText('Submit')).toBeNull()
    spy.mockRestore();
  })*/

  it("should not crash if missing employee info", async () => {
    const spy = jest.spyOn(console, errorState).mockReturnValue()
    sessionStorage.clear()
    await act(async () => {
      render(<CaseHistory />)
    })
    expect(screen.queryByText('Mock Tester')).toBeNull();
    expect(screen.queryByText('test@mock.com')).toBeNull();
    expect(screen.queryByText('mock1')).toBeNull();
    expect(screen.queryByText('"2021-10-07"')).toBeNull();
    spy.mockRestore();
  })
})
