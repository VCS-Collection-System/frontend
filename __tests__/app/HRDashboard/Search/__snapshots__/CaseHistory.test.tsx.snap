// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CaseHistory /> should match existing snapshot 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card pf-m-rounded"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__title"
    >
      <p
        class=""
        data-pf-content="true"
      >
        Case History
      </p>
      <div
        class="pf-c-content"
      >
        <small
          class=""
          data-pf-content="true"
        >
          Review Case History
        </small>
      </div>
    </div>
    <div
      class="pf-c-card__body"
    >
      <div
        class="pf-l-grid pf-m-gutter"
      >
        <div
          class="pf-l-grid__item pf-m-12-col"
        >
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-2"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__title"
            >
              <p
                class=""
                data-pf-content="true"
              >
                Employee Information
              </p>
              <div
                class="pf-c-content"
              >
                <small
                  class=""
                  data-pf-content="true"
                />
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              <form
                class="pf-c-form"
                novalidate=""
              >
                <div
                  class="pf-l-grid pf-m-gutter"
                >
                  <div
                    class="pf-l-grid__item pf-m-6-col"
                  >
                    <div
                      class="pf-c-form__group"
                    >
                      <div
                        class="pf-c-form__group-label"
                      >
                        <label
                          class="pf-c-form__label"
                          for="profile-form-employee-id"
                        >
                          <span
                            class="pf-c-form__label-text"
                          >
                            Person ID
                          </span>
                        </label>
                         
                      </div>
                      <div
                        class="pf-c-form__group-control"
                      >
                        <input
                          aria-invalid="false"
                          aria-label="profile-input-employee-id"
                          class="pf-c-form-control"
                          disabled=""
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="pf-c-form__group"
                    >
                      <div
                        class="pf-c-form__group-label"
                      >
                        <label
                          class="pf-c-form__label"
                          for="profile-form-employee-name"
                        >
                          <span
                            class="pf-c-form__label-text"
                          >
                            Name
                          </span>
                        </label>
                         
                      </div>
                      <div
                        class="pf-c-form__group-control"
                      >
                        <input
                          aria-invalid="false"
                          aria-label="profile-input-employee-name"
                          class="pf-c-form-control"
                          disabled=""
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="pf-l-grid__item pf-m-6-col"
                  >
                    <div
                      class="pf-c-form__group"
                    >
                      <div
                        class="pf-c-form__group-label"
                      >
                        <label
                          class="pf-c-form__label"
                          for="profile-form-employee-email"
                        >
                          <span
                            class="pf-c-form__label-text"
                          >
                            Email
                          </span>
                        </label>
                         
                      </div>
                      <div
                        class="pf-c-form__group-control"
                      >
                        <input
                          aria-invalid="false"
                          aria-label="profile-input-employee-email"
                          class="pf-c-form-control"
                          disabled=""
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </article>
        </div>
        <div
          class="pf-l-grid__item pf-m-4-col pf-m-6-row"
        >
          <article
            class="pf-c-card"
            data-ouia-component-id="OUIA-Generated-Card-3"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            id=""
          >
            <div
              class="pf-c-card__title"
            >
              <p
                class=""
                data-pf-content="true"
              >
                Attestation Records
              </p>
              <div
                class="pf-c-content"
              >
                <small
                  class=""
                  data-pf-content="true"
                />
              </div>
            </div>
            <div
              aria-label="Global"
              class="pf-c-menu"
              data-ouia-component-id="OUIA-Generated-Menu-1"
              data-ouia-component-type="PF4/Menu"
              data-ouia-safe="true"
            >
              <div
                class="pf-c-menu__content"
                style="--pf-c-menu__content--Height: 875px;"
              >
                <section
                  class="pf-c-menu__group"
                >
                  <h1
                    class="pf-c-menu__group-title"
                    id=""
                  >
                    Test Results
                  </h1>
                  <ul
                    class="pf-c-menu__list indentList"
                  >
                    <li
                      class="pf-c-menu__list-item pf-m-disabled"
                      tabindex="-1"
                    >
                      <button
                        aria-current="false"
                        class="pf-c-menu__item"
                        disabled=""
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="pf-c-menu__item-main"
                        >
                          <span
                            class="pf-c-menu__item-text"
                          >
                            [No Test Results]
                          </span>
                        </span>
                      </button>
                    </li>
                  </ul>
                </section>
                <section
                  class="pf-c-menu__group"
                >
                  <h1
                    class="pf-c-menu__group-title"
                    id=""
                  >
                    Vaccination Records
                  </h1>
                  <ul
                    class="pf-c-menu__list indentList"
                  >
                    <li
                      class="pf-c-menu__list-item pf-m-disabled"
                      tabindex="-1"
                    >
                      <button
                        aria-current="false"
                        class="pf-c-menu__item"
                        disabled=""
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="pf-c-menu__item-main"
                        >
                          <span
                            class="pf-c-menu__item-text"
                          >
                            [No Vaccination Records]
                          </span>
                        </span>
                      </button>
                    </li>
                  </ul>
                </section>
              </div>
            </div>
          </article>
        </div>
        <div
          class="pf-l-grid__item pf-m-8-col pf-m-6-row"
        >
          <article
            class="pf-c-card pf-m-rounded"
            data-ouia-component-id="OUIA-Generated-Card-4"
            data-ouia-component-type="PF4/Card"
            data-ouia-safe="true"
            height="500px"
            id=""
          >
            <div
              class="pf-c-card__title"
            >
              <p
                class=""
                data-pf-content="true"
              >
                Attestation Details
              </p>
              <div
                class="pf-c-content"
              >
                <small
                  class=""
                  data-pf-content="true"
                />
              </div>
            </div>
            <div
              class="pf-c-card__body"
            >
              Please Select an Item
            </div>
            <div
              class="pf-c-card__footer"
            >
              <div
                class="pf-l-split pf-m-gutter"
              />
            </div>
          </article>
        </div>
      </div>
    </div>
    <div
      class="pf-c-card__footer"
    >
      <div
        class="pf-l-split pf-m-gutter"
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-primary"
          data-ouia-component-id="OUIA-Generated-Button-primary-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          Add Record
        </button>
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-plain"
          data-ouia-component-id="OUIA-Generated-Button-plain-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          Back
        </button>
      </div>
    </div>
  </article>
</DocumentFragment>
`;
